### PR TITLE
eclipse downloads for some reason removed the http prefix from the do…

### PR DESCRIPTION
**Pull request by pellcorp**
…wnload urls

### CODE REVIEW

File: umake/frameworks/ide.py [hunk 1]
```diff
@@ -93,9 +93,9 @@ def parse_download_link(self, line, in_download):
         else:
             in_download = False
         if in_download:
-            p = re.search(r"href='(http://www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'", line)
+            p = re.search(r"href='(//www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'", line)
```
Comment (LyzardKing) [inline umake/frameworks/ide.py]:
```suggestion
            p = re.search(r"href='(http:|https:)?(//www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'", line)
```

I'd change the regex so it works with and without the https prefix.
The checksum pages seem to all be available via https, so we don't need to check that.

File: umake/frameworks/ide.py [hunk 2]
```diff
@@ -93,9 +93,9 @@ def parse_download_link(self, line, in_download):
...
             with suppress(AttributeError):
-                self.new_download_url = p.group(1).replace('download.php', 'sums.php').replace('http://', 'https://')
+                self.new_download_url = 'https:' + p.group(1).replace('download.php', 'sums.php').replace('http://', 'https://')
                 self.https = True if parse.splittype(self.new_download_url)[0] is "https" else False
```
Comment (LyzardKing) [inline umake/frameworks/ide.py]: This line can be removed
The lines...107-108 with the if can be removed

File: umake/frameworks/ide.py [hunk 3]
```diff
@@ -93,9 +93,9 @@ def parse_download_link(self, line, in_download):
         else:
             in_download = False
         if in_download:
-            p = re.search(r"href='(http://www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'", line)
+            p = re.search(r"href='(//www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'", line)
             with suppress(AttributeError):
-                self.new_download_url = p.group(1).replace('download.php', 'sums.php').replace('http://', 'https://')
+                self.new_download_url = 'https:' + p.group(1).replace('download.php', 'sums.php').replace('http://', 'https://')
```
Comment (LyzardKing) [inline umake/frameworks/ide.py]:
```suggestion
                self.new_download_url = 'https:' + p.group(2).replace('download.php', 'sums.php')
```

**Comment by pellcorp** [comment]
I am working on the changes

**Comment by pellcorp** [comment]
still fails due to line length issues, which is weird because when I open the file its not 128 characters long.  Anyway I don't have a dev env setup for ubuntu-mate but when I have some time I will so I can try and get the PR merged.

**Comment by LyzardKing** [comment]
With the changes i suggested it seems to work even in the tests, so let me know if there's something you need to verify

**Comment by LyzardKing** [comment]
The line too long issue is because you need to split it like this:
```
            p = re.search(r"href='(http:|https:)?(//www\.eclipse\.org\/downloads/download\.php\?file=.*\.tar\.gz)'",
                          line)
```

**Comment by pellcorp** [comment]
yay

**Comment by LyzardKing** [comment]
It looks good!
I'll just run some extra tests on the other eclipse installers, but it should be fine.
Thanks!

**Pull request merged**